Client/WarFare/UIHotKeyDlg.cpp

```diff
@@ -976,7 +975,10 @@ bool CUIHotKeyDlg::SetReceiveSelectedItem(int iIndex)
 		spSkill->pUIIcon->SetMoveRect(pArea->GetRegion());
 	}
 	
-	m_pMyHotkey[m_iCurPage][iIndex] = spSkill;
+	if (m_pMyHotkey[m_iCurPage][iIndex] == nullptr)
```

Comment:
I think you misunderstood. You just swapped the potential memory leak from one side to the other with this check; now spSkill (and pUIIcon) will leak, rather than the original icon that you were blindly replacing.

As I suggested, add the preventative check to the start - there's no reason for it to be set in the first place, so you can just pre-check it at the start of the method and early fail.